### DISCUSSION TITLE
How can I convert string 16 to string 64?

### DISCUSSION THREAD

**Discussion by nihilitysoul** [bpftrace/bpftrace#2238] [Q&A] [answered]
I create a map that its key is string 64, but I want to put comm as a key, while comm is string 16, how can I convert comm to string 64?

My code like that:
BEGIN
{
  @list["test"] = 1;
}

tracepoint:syscalls:sys_enter_open
/@list[comm]/
{
...
}

The result is 

```
ERROR: Argument mismatch for @list: trying to access with arguments: [string[16]] when map expects arguments: [string[64]]
/@list[comm]/
~~~~~~~~~~~~~~
```

Please help me...

**Comment by viktormalik** [comment, marked as answer]
This problem was fixed by #2044 which was a part of the bpftrace 0.14.0 release. So I'd suggest updating your bpftrace and it should work.

**Reply by nihilitysoul** [reply]
Thanks for your answer, it's very useful to me.

**Comment by hooklab** [comment]
why my bpftrace still like this.

[root@ying tmp]# bpftrace -V
bpftrace v0.14.1

**uname -a** 
Linux ying 5.18.5-200.fc36.x86_64 #1 SMP PREEMPT_DYNAMIC Thu Jun 16 14:51:11 UTC 2022 x86_64 x86_64 x86_64 GNU/Linux

BEGIN
{
    @list["rm"] = 1;
    printf("#start fs change monitor...\n");
}

kprobe:vfs_rmdir,
kprobe:vfs_unlink
/@list[comm]/
{

    print("test");
    exit();
}

scripts/fs.bt:19:1-13: ERROR: Argument mismatch for @list: trying to access with arguments: [string[16]] when map expects arguments: [string[64]]
/@list[comm]/
~~~~~~~~~~~~

**Reply by hooklab** [reply]
@nihilitysoul

**Reply by viktormalik** [reply]
Could you provide output of `bpftrace --info` please? And also the exact command which you use and the complete contents of `fs.bt`? Thanks!

**Reply by hooklab** [reply]
> Could you provide output of `bpftrace --info` please? And also the exact command which you use and the complete contents of `fs.bt`? Thanks!

**[root@ying tmp]# bpftrace --info**
System
  OS: Linux 5.18.5-200.fc36.x86_64 #1 SMP PREEMPT_DYNAMIC Thu Jun 16 14:51:11 UTC 2022
  Arch: x86_64

Build
  version: v0.14.1
  LLVM: 14.0.0
  ORC: v2
  foreach_sym: yes
  unsafe uprobe: no
  bfd: yes
  bpf_attach_kfunc: yes
  bcc_usdt_addsem: yes
  bcc bpf_attach_uprobe refcount: yes
  bcc library path resolution: yes
  libbpf: yes
  libbpf btf dump: yes
  libbpf btf dump type decl: yes
  libdw (DWARF support): no

Kernel helpers
  probe_read: yes
  probe_read_str: yes
  probe_read_user: yes
  probe_read_user_str: yes
  probe_read_kernel: yes
  probe_read_kernel_str: yes
  get_current_cgroup_id: yes
  send_signal: yes
  override_return: no
  get_boot_ns: yes
  dpath: yes

Kernel features
  Instruction limit: 1000000
  Loop support: yes
  btf (depends on Build:libbpf): yes
  map batch (depends on Build:libbpf): yes
  uprobe refcount (depends on Build:bcc bpf_attach_uprobe refcount): yes

Map types
  hash: yes
  percpu hash: yes
  array: yes
  percpu array: yes
  stack_trace: yes
  perf_event_array: yes

Probe types
  kprobe: yes
  tracepoint: yes
  perf_event: yes
  kfunc: yes
  iter:task: yes
  iter:task_file: yes


[root@ying tmp]# **bpftrace --unsafe --no-warnings  scripts/fs.bt**
scripts/fs.bt:19:1-13: ERROR: Argument mismatch for @list: trying to access with arguments: [string[16]] when map expects arguments: [string[64]]
/@list[comm]/
~~~~~~~~~~~~

**Reply by viktormalik** [reply]
Are you building bpftrace from source?

**Reply by hooklab** [reply]
> Are you building bpftrace from source?

no 

**sudo dnf list bpftrace**


bpftrace.x86_64                                                  0.14.1-1.fc36                                                  @updates
